imap-proto/src/types.rs

```diff
@@ -111,13 +111,101 @@ pub enum SectionPath {
     Part(Vec<u32>, Option<MessageSection>),
 }
 
+#[derive(Debug, Eq, PartialEq)]
+pub struct BodyParam<'a> {
```

Comment:
@djc Would you prefer body param as is here, or as a tuple?

Reply:
I think a tuple is probably fine for this.